src/eth/Cdp.js

```diff
@@ -6,6 +6,7 @@ export default class Cdp {
   constructor(cdpService, cdpId = null) {
     this._cdpService = cdpService;
     this._smartContractService = this._cdpService.get('smartContract');
+    this._nonceService = this._cdpService.get('nonce');
```

Comment:
you could also get this from `transactionManager` and then you wouldn't have to depend upon `nonce` directly... in fact, now that i look at it, you could use `transactionManager.formatHybridTx` below and then not have to use the nonce service directly at all

Reply:
The problem with this approach is we can't wait for `contractPromise` to resolve if we use `formatHybridTx`, since the contract promise then doesn't exist in this scope. So waiting for it would be equivalent to awaiting `onMined`